### PR TITLE
refactor: remove FunctionComponent type

### PR DESCRIPTION
**Pull request by lja1018**
### Please check if the PR fulfills these requirements
- [x] It's submitted to right branch according to our branching model
- [x] It's right issue type on title
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has description for the breaking change

### Description

- remove `FunctionComponent` type
  - use `Props` directly. If use children of props, use `PropsWithChildren` utility type
- change function expressions to function declarations

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

**Review by adhrinae — APPROVED** [review]
리뷰 완료합니다.

File: apps/calendar/src/components/popup/dateSelector.tsx [hunk 1]
```diff
@@ -27,96 +27,97 @@ const classNames = {
...
 };
 
-export const DateSelector = forwardRef<DateRangePicker, Props>(
-  ({ start, end, isAllday = false, formStateDispatch }, ref) => {
-    const { usageStatistics } = useStore(optionsSelector);
-    const startPickerContainerRef = useRef<HTMLDivElement>(null);
-    const startPickerInputRef = useRef<HTMLInputElement>(null);
-    const endPickerContainerRef = useRef<HTMLDivElement>(null);
-    const endPickerInputRef = useRef<HTMLInputElement>(null);
+export const DateSelector = forwardRef<DateRangePicker, Props>(function DateSelector(
```
Comment (jajugoguma) [inline apps/calendar/src/components/popup/dateSelector.tsx:30]: 화살표 함수에서 기명함수로 변경된 이유는 무엇인가요??

**Review by jajugoguma — APPROVED** [review]
리뷰 완료합니다!

File: apps/calendar/src/components/popup/dateSelector.tsx (hunk 1 above)
Comment (adhrinae) [reply apps/calendar/src/components/popup/dateSelector.tsx:30]: Devtool에서 이름이 바로 나옵니다. 상황에 따라서 HoC(고차 컴포넌트)로 감싸진 컴포넌트는 익명 컴포넌트로 처리되거나 이름이 제대로 표시되지 않아 별도의 처리를 해주어야 하는 경우가 있어서요.

Comment (jajugoguma) [reply apps/calendar/src/components/popup/dateSelector.tsx:30]: @adhrinae 
아하, 감사합니다!

**Pull request merged**